File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.go
func peakIndexInMountainArray(arr []int) int {
    n:=len(arr)
    low:=0
    high:=n-1
    res:=-1

    for low<=high{
        guess:=(low+high)/2

        if arr[guess]<arr[guess+1]{
            low=guess+1
        }else{
            res=guess
            high=guess-1
        }
    }
    return res
}

File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var peakIndexInMountainArray = function(arr) {
    let n=arr.length;
    let low=0;
    let high=n-1;
    let res=-1;

    while(low<=high){
        let guess=Math.floor((low+high)/2);

        if(arr[guess]<arr[guess+1]){
            low=guess+1;
        }
        else{
            res=guess;
            high=guess-1;
        }
    }
    return res;
};